R/get_predictor_data.R

```diff
@@ -0,0 +1,37 @@
+#' Extract a data.frame of a model's predictors
+#'
+#' This internal function extract predictors from a model's formula and extracts
+#' the corresponding columns from a data.frame.
+#'
+#' @noRd
+#'
+#' @author Thibaut Jombart
+#'
+#' @param model A [`trending_model()`] object.
+#'
+#' @param df A `data.frame` containing data required by the models (and possibly
+#'   non-relevant columns as well).
+#'
+#' @return A `data.frame` containing exactly the predictors of the model; if
+#'   some data are missing, an error is issued.
+#'
+get_predictor_data <- function(model, df) {
+  ## Extract variable names from the formula
+  form <- trending::get_formula(model)
+  vars <- unlist(strsplit(as.character(form)[3], "[-:+*|]"))
```

Comment:
would something like `all.vars(form)[-1]` be better here?

Reply:
Nice! Will push the change